master/internal/api/handler.go

```diff
@@ -3,13 +3,54 @@ package api
 import (
 	"fmt"
 	"net/http"
+	"net/url"
+	"unicode/utf8"
 
 	"github.com/gorilla/websocket"
 	"github.com/labstack/echo/v4"
 	"github.com/pkg/errors"
 )
 
-var upgrader = websocket.Upgrader{}
+// equalASCIIFold returns true if s is equal to t with ASCII case folding as
+// defined in RFC 4790.
+func equalASCIIFold(s, t string) bool {
```

Comment:
out of curiosity... what is wrong with [Strings.ToLower()](https://pkg.go.dev/strings#ToLower)?